buildpacks/java/tests/build_test.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2021-2022 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package tests
+
+import (
+	"os"
+	"testing"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	. "github.com/onsi/gomega"
+
+	"kn-fn/java-function-buildpack/java"
+)
+
+const (
+	HTTPFuncTemplate = `from typing import Any
+
+def {{.Name}}({{range $var, $type := .Arguments}}{{$var}}:{{$type}}{{end}}):
+	return "{{.ReturnValue}}"`
+)
+
+func TestBuild(t *testing.T) {
+	spec.Run(t, "Build", testBuild, spec.Report(report.Terminal{}))
+}
+
+func testBuild(t *testing.T, when spec.G, it spec.S) {
+	var Expect = NewWithT(t).Expect
+
+	var (
+		build         java.Build
+		cleanupAppDir func()
+		context       libcnb.BuildContext
+	)
+
+	it.Before(func() {
+		build = java.Build{
+			Logger: NewLogger(),
+		}
+	})
+
+	it.After(func() {
+		cleanupAppDir()
+	})
+
+	when("#Build", func() {
+		var result libcnb.BuildResult
+
+		it.Before(func() {
+			var (
+				appDir string
+				err    error
+			)
+
+			Expect(os.Setenv("BP_FUNCTION", "some_module.some_function"))
+			appDir, cleanupAppDir = SetupTestDirectory(
+				WithFuncYaml(),
+				WithFunctionFile("some_module", "some_function", HTTPFuncTemplate),
+			)
+
+			context = makeBuildContext(
+				withBuildApplicationPath(appDir),
+				withDependencies([]map[string]any{
+					{"id": "invoker-deps", "version": "1.2.3"},
+					{"id": "invoker", "version": "2.3.4"},
+				}),
+				withOptions(map[string]any{
+					"some-option":       "some-value",
+					"some-other-option": "some-other-value",
+				}),
+			)
+
+			result, err = build.Build(context)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		it.After(func() {
+			Expect(os.Unsetenv("BP_FUNCTION"))
+		})
+
+		it("adds expected layers", func() {
+			var layers []string
+			for _, l := range result.Layers {
+				layers = append(layers, l.Name())
+			}
+
+			Expect(layers).To(Equal([]string{
+				"java-function",
+				"invoker",
+			}))
+		})
+
+		it("adds expected labels", func() {
+			Expect(result.Labels).To(Equal([]libcnb.Label{
```

Comment:
This will flake on occasion if you're not sorting the labels in `build.go`. See [python for an example](https://github.com/vmware-tanzu/function-buildpacks-for-knative/pull/126/files#diff-43e6a86218a07ba3c7203de725ce20775d400bcd91995206ac35abd34aadeddfR91).

Reply:
+1 let's add the Sort to match behavior

Reply:
Sorted the labels in build.go and modified the hardcoded values in test to validate that sort is happening